crates/tabby/src/main.rs

```diff
@@ -124,29 +133,54 @@ async fn main() {
 
     let config = Config::load().unwrap_or_default();
 
-    match &cli.command {
-        Commands::Serve(args) => serve::main(&config, args).await,
-        Commands::Download(args) => download::main(args).await,
-        Commands::Scheduler(args) => tabby_scheduler::scheduler(args.now, &config)
+    match cli.command {
+        Commands::Serve(ref args) => serve::main(&config, args).await,
+        Commands::Download(ref args) => download::main(args).await,
+        Commands::Scheduler(args) => tabby_scheduler::scheduler(args.now, config.repositories)
             .await
             .unwrap_or_else(|err| fatal!("Scheduler failed due to '{}'", err)),
         #[cfg(feature = "ee")]
-        Commands::JobSync => tabby_scheduler::job_sync(&config),
+        Commands::JobSync(args) => {
+            let repositories = get_repositories(args.url, args.token, &config.repositories);
+            tabby_scheduler::job_sync(&repositories)
+        }
         #[cfg(feature = "ee")]
-        Commands::JobIndex => tabby_scheduler::job_index(&config),
+        Commands::JobIndex(args) => {
+            let repositories = get_repositories(args.url, args.token, &config.repositories);
+            tabby_scheduler::job_index(&repositories)
+        }
         #[cfg(feature = "ee")]
-        Commands::WorkerCompletion(args) => {
+        Commands::WorkerCompletion(ref args) => {
             worker::main(tabby_webserver::public::WorkerKind::Completion, args).await
         }
         #[cfg(feature = "ee")]
-        Commands::WorkerChat(args) => {
+        Commands::WorkerChat(ref args) => {
             worker::main(tabby_webserver::public::WorkerKind::Chat, args).await
         }
     }
 
     opentelemetry::global::shutdown_tracer_provider();
 }
 
+fn get_repositories(
```

Comment:
this function should be implemented in a seaprate file, e.g `job.rs`

Like what we did for `worker::main`. I recommend extract an `job::main` as well.

Reply:
I extracted this function to another file, but I don't think a main function in here makes sense, since each of the two jobs are initialized in two lines of code (and could be inlined into one).